tests/robustness/model/deterministic.go

```diff
@@ -351,15 +351,17 @@ type LeaseRevokeRequest struct {
 }
 type DefragmentRequest struct{}
 
-// MaybeEtcdResponse extends EtcdResponse to represent partial or failed responses.
-// Possible states:
-// * Normal response. Only EtcdResponse is set.
-// * Partial response. The EtcdResponse.Revision and PartialResponse are set.
-// * Failed response. Only Err is set.
+// MaybeEtcdResponse extends EtcdResponse to include partial information about responses to a request.
+// Possible response state information:
+// * Normal response. Client observed response. Only EtcdResponse is set.
+// * Persisted. Client didn't observe response, but we know it was persisted by etcd. Only Persisted is set
```

Comment:
Q. So, meaning, the PUT was finalized on the consensus + storage, but response wasn't received from a client point of view? Seems confusing, given the function above `describeEtcdResponse` strings this as "unknown"? Should we make the naming more consistent?

Reply:
A great question, I have been thinking about changing the naming here. 

> So, meaning, the PUT was finalized on the consensus + storage, but response wasn't received from a client point of view?

Yes, from client perspective it got error so we don't know the response, but we know that request was persisted thanks to looking up WAL.

> Seems confusing, given the function above describeEtcdResponse strings this as "unknown"? Should we make the naming more consistent?

The naming in robustness tests is constantly evolving based on feedback. Name "unknown" was meant to just mean we don't know the exact response but we knew it's revision due to matching watch response.  Name "persisted" was introduced only recently into code with PR that added reading the WAL. 

I think it would make sense to rename "unknown" to "persisted" in describeEtcdResponse. Does that sound good for everyone? @MadhavJivrajani @siyuanfoundation @fuweid

Reply:
Thanks for the explanation. "persisted" makes sense to me. Defer to other contributors :)

Reply:
"persisted" makes sense to me

Reply:
+1 to "persisted"

Reply:
Renaming to 'persisted' makes sense in the context of WAL. 

I am curious aren't there any edge cases where 'persisted' might not fully capture the state, or scenarios where additional qualifiers might be necessary?

Reply:
> I am trying to understand could there be any edge case where 'persisted' might not fully capture the state, or scenarios where additional qualifiers might be necessary?

Good question, I think the second case would be when request generates a event and we observe it on watch. This is where the "PersistedRevision" comes from. So name "persisted" might not be capturing all cases. Maybe "observed" would be capturing the whole meaning. Observed not by client, but by other side effects, like WAL or WATCH. Don't know if "observed" is better than "persisted" though. 

Let's not block this PR on picking the name. If someone is interested feel free to file a separate PR to propose and discuss a new name.